extension/src/validator/validator-builder.js

```diff
@@ -6,6 +6,15 @@ function withPayment(paymentObject) {
   const errors = {}
 
   return {
+    validateRequiredFields() {
+      if (!paymentObject.custom.fields.commercetoolsProjectKey)
```

Comment:
what if `paymentObject.custom` is undefined in the request, it would cause NPE, maybe check other similar validation method calls.

Also what if the value is there but it has whiteSpace chars? Should it be a warning too?

Reply:
Good point 👍 I think for whiteSpace it should also fail...

Reply:
c0fc985

Reply:
please validate all in once not step by step 😉 

with looking at the code, it also might be there is no way to put undefined or null into the paymentObject.custom 🤔 or even to fields could have some field if they are empty (not sure) maybe you need to check what API returns.

Also, whiteSpace check should not be checked that way, those following texts are also not valid: "projectName ", " projectName", "project name". maybe check if the string has an empty string?

I'm also not sure from that please check both ct and adyen docs to ensure the possible formats, the benefit is failing fast that way so we could avoid extra calls and exception handling in the extension module.

Reply:
So, paymentDraft.custom is optional - so it could be undefined or null. If custom is provided but not fields API adds empty fields.

Reply:
Quickly checked the docs; could not find anything about the formats other than both CT and adyen are generating these values - as they need to have url conformity i assume whitespaces are not allowed, lead to errors.

Reply:
9f61a99

Reply:
Actually this is the requirement for the project key in CT: `May only contain between 2 and 36 alphanumeric characters, underscores, or hyphens (no white space or special characters like ñ, ü, #, %).`